test/multiplexed-session.ts

```diff
@@ -0,0 +1,286 @@
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+      clock = sandbox.useFakeTimers();
+    });
+    afterEach(() => {
+      clock.restore();
+    });
+    it('should set an interval to refresh mux sessions', () => {
+      const expectedInterval =
+        multiplexedSession.refreshRate! * 24 * 60 * 60000;
+
+      multiplexedSession._maintain();
+      clock.tick(expectedInterval);
+      assert.strictEqual(createSessionStub.callCount, 1);
+    });
+  });
+
+  describe('_createSession', () => {
+    it('should create the mux sessions with multiplexed option', async () => {
+      await multiplexedSession._createSession();
+      assert.strictEqual(createSessionStub.callCount, 1);
+      assert.deepStrictEqual(createSessionStub.lastCall.args[0], {
+        multiplexed: true,
+      });
+    });
+
+    it('should reject with any request errors', async () => {
+      const error = new Error('err');
+      createSessionStub.rejects(error);
+
+      try {
+        await multiplexedSession._createSession();
+        throw new Error('Should not make it this far.');
+      } catch (e) {
+        assert.strictEqual(e, error);
+      }
+    });
+  });
+
+  describe('getSession', () => {
+    it('should acquire a session', done => {
+      sandbox.stub(multiplexedSession, '_acquire').resolves(fakeMuxSession);
+
+      multiplexedSession.getSession((err, session) => {
+        assert.ifError(err);
+        assert.strictEqual(session, fakeMuxSession);
+        done();
+      });
+    });
+
+    it('should pass any errors to the callback', done => {
+      const error = new Error('err');
+      sandbox.stub(multiplexedSession, '_acquire').rejects(error);
+      multiplexedSession.getSession(err => {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should pass back the session and txn', done => {
+      const fakeTxn = new FakeTransaction() as unknown as Transaction;
+
+      fakeMuxSession.txn = fakeTxn;
+
+      sandbox.stub(multiplexedSession, '_acquire').resolves(fakeMuxSession);
+
+      multiplexedSession.getSession((err, session, txn) => {
+        assert.ifError(err);
+        assert.strictEqual(session, fakeMuxSession);
+        assert.strictEqual(txn, fakeTxn);
+        done();
+      });
+    });
+  });
+
+  describe('_acquire', () => {
+    it('should return a session', async () => {
+      sandbox.stub(multiplexedSession, '_getSession').resolves(fakeMuxSession);
+      const session = await multiplexedSession._acquire();
+      assert.strictEqual(session, fakeMuxSession);
+    });
+
+    it('should have the multiplexed property set to true', async () => {
+      sandbox.stub(multiplexedSession, '_getSession').resolves(fakeMuxSession);
+      const session = await multiplexedSession._acquire();
+      assert.strictEqual(session.multiplexed, true);
+      assert.strictEqual(fakeMuxSession.multiplexed, true);
+    });
+  });
+
+  describe('_getSession', () => {
```

Comment:
Since, we are listening to the error event in _getSession in case of the error from the _createSession, I have added a test for the case when the error event will get emitted, if we are able to listen to that event in _getSession